content/ADR-151.md

```diff
@@ -0,0 +1,29 @@
+---
+layout: adr
+adr: 151
+title: Rendering order for UI entities
+date: 2022-12-14
+status: Review
+type: Standards Track
+spdx-license: CC0-1.0
+authors:
+  - menduz
+---
+
+## Abstract
+
+This document describes the requirements to render the UI entities for the Decentraland Protocol. TL;DR; Pre-order using the entity tree defined by the UiTransform ([ADR-124](/adr/ADR-124)).
+
+## Specification
+
+The UI entities are derived from a synthetic tree of entities defined by the UiTransformComponent ([ADR-124](/adr/ADR-124)). That tree should be rendered from the root to the leafs recursively using a pre-order sequencing.
```

Comment:
The specification re-written by Chat GPT (just for fun as well):
```
The UI entities in this system are derived from a synthetic tree of entities defined by the UiTransformComponent ([ADR-124](https://chat.openai.com/adr/ADR-124)). This tree must be rendered using a pre-order sequencing, with children overwriting the pixels of the parent in the rendering process. This behavior is similar to how HTML, ReactNative, and Unity's UiToolkit handle rendering by default.

It is necessary to be able to configure entities to act as stencils, so that only the content within the borders of the parent entity is rendered. By default, children must be rendered in full (without stencils), even if they extend beyond the borders of the parent. This stencil behavior is analogous to the overflow: hidden CSS property.
```